qos/jsonrpc/request.go

```diff
@@ -129,3 +129,37 @@ func BuildParamsFromObjectAndString(objectParam map[string]string, stringParam s
 	}
 	return Params{rawMessage: jsonParams}, nil
 }
+
+// BuildParamsFromStringAndObject builds a Params object from a single string and a map.
+//
+// For example, for a Solana `getSignaturesForAddress` request, the params would look like:
+// params - ["Vote111111111111111111111111111111111111111",{"limit":1}]
+//
+// Used for getSignaturesForAddress
+func BuildParamsFromStringAndObject(stringParam string, objectParam map[string]any) (Params, error) {
```

Comment:
A single method on Request, e.g. SetParams([]byte), should be sufficient.
These special case methods can then live in the client code.

Reply:
Added `TODO_TECHDEBT`.